plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py

```diff
@@ -17,149 +25,52 @@
-
-    Returns
-    -------
-    Parameter or str
-        A Parameter if the string fitted one of the definitions, else the input string
-    """
-    # lazy initialization to avoid overhead when loading hydra
-    import nevergrad as ng
-
-    # revert config parsing
-
-    if isinstance(description, (ListConfig, list)):
-        description = ",".join(str(x) for x in description)
-    if isinstance(description, str):
-        # cast to spec if possible
-        try:
-            description = CommandlineSpec.parse(description)
-        except ValueError:
-            pass
-    # convert scalar commandline specs to dict
-    if isinstance(description, CommandlineSpec) and description.bounds is not None:
-        description = ScalarConfigSpec(
-            lower=description.bounds[0],
-            upper=description.bounds[1],
-            log=description.log,
-            integer=description.cast == "int",
-        )
-    # convert scalar config specs to dict
-    # convert dict to Scalar parameter instance
+def create_nevergrad_parameter(description: Any) -> Any:
+    scalar = None
+    if isinstance(description, Override):
+        override = description
+        val = override.value()
+        if override.is_sweep_override():
+            if override.is_choice_sweep():
+                val = cast(ChoiceSweep, val)
+                vals = [
+                    x for x in override.sweep_iterator(transformer=Transformer.encode)
+                ]
+                if "ordered" in val.tags:
+                    return ng.p.TransitionChoice(vals)
+                else:
+                    return ng.p.Choice(vals)
+            elif override.is_range_sweep():
+                # if shuffled or integer range and <=6, use Choice
+                val = cast(RangeSweep, val)
+                if val.shuffle or (
+                    val.stop is not None
+                    and val.start is not None
+                    and val.stop - val.start <= 6
+                ):
+                    vals = [
+                        x
+                        for x in override.sweep_iterator(transformer=Transformer.encode)
+                    ]
+                    return ng.p.Choice(vals)
+                params = {"lower": val.start, "upper": val.stop}
+                scalar = ng.p.Scalar(**params)  # type: ignore
+                if isinstance(val.start, int):
+                    scalar.set_integer_casting()
+            elif override.is_interval_sweep():
+                val = cast(IntervalSweep, val)
+                if "log" in val.tags:
+                    scalar = ng.p.Log(lower=val.start, upper=val.end)
+                else:
+                    scalar = ng.p.Scalar(lower=val.start, upper=val.end)
+                if isinstance(val.start, int):
+                    scalar.set_integer_casting()
+        else:
+            return val
+    if isinstance(description, (list, ListConfig)):
+        return ng.p.Choice(list(description))
```

Comment:
if you need to convert ListConfig to list use OmegaConf.to_container(description, resolve=True)

Reply:
got it, thx for the tip!